django/core/management/commands/shell.py

```diff
@@ -44,21 +45,35 @@ def add_arguments(self, parser):
             ),
         )
 
+    def display_message(self, func):
```

Comment:
We don’t need to use a decorator-style wrapper like this. Instead, make two methods:

```python
def get_and_report_namespace(self):
    namespace = self.get_namespace()
    self.stdout.write(...)
    return namespace

def get_namespace(self):
    ...
```

That’s what I meant by wrapper, not necessarily a decorator wrapper.

Also we will want to avoid the message when verbosity=0